src/lib/Behat/BrowserContext/UserNotificationContext.php

```diff
@@ -47,4 +51,133 @@ public function iGoToUserNotificationWithDetails(TableNode $notificationDetails)
         $this->userNotificationPopup->verifyIsLoaded();
         $this->userNotificationPopup->clickNotification($type, $description);
     }
+
+    /**
+     * @Then the notification appears with details:
+     */
+    public function thereIsNotificationAppearsWithDetails(TableNode $notificationDetails): void
+    {
+        $type = $notificationDetails->getHash()[0]['Type'];
+        $author = $notificationDetails->getHash()[0]['Author'];
+        $description = $notificationDetails->getHash()[0]['Description'];
+        $date = $notificationDetails->getHash()[0]['Date'];
+
+        $this->userNotificationPopup->verifyIsLoaded();
+        $this->userNotificationPopup->verifyNotification($type, $author, $description, $date, true);
+    }
+
+    /**
+     * @When I open notification menu with description :description
+     */
+    public function iOpenNotificationMenu(string $description): void
+    {
+        $this->userNotificationPopup->openNotificationMenu($description);
+    }
+
+    /**
+     * @When I click :action action
+     */
+    public function iClickActionButton(string $action): void
+    {
+        $this->userNotificationPopup->clickActionButton($action);
+    }
+
+    /**
+     * @Then the notification action should be :expectedAction
+     */
+    public function verifyNotificationAction(string $expectedAction): void
+    {
+        $this->userNotificationPopup->getActionButton($expectedAction);
+    }
+
+    /**
+     * @When I click mark all as read button
+     */
+    public function iClickButton(): void
+    {
+        $this->userNotificationPopup->verifyIsLoaded();
+        $this->userNotificationPopup->clickOnMarkAllAsReadButton();
+    }
+
+    /**
+     * @When I click view all notifications button
```

Comment:
Same as workflow repo comment - description should be more verbose here, with intent stated.